enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,220 @@
+
+Timeline:  ~60 minutes
+
+60 minutes?  That seems high!  That's because it is high.  We want to reduce
+the noise.  We've done a lot to make clusters and operators auto-heal
+themselves.  The whole idea is that if a condition has persisted for more than
+60 minutes, it's unlikely to be resolved without intervention any time later.
+
+#### Example 1: All machine-config-daemons are crashlooping.
+
+That seems significant
+and it probably is, but nobody has to get out of bed to fix this **right now**.
+But, what if a machine dies, I won't be able to get a replacement?  Yeah, that
+is probably true, but how often does that happen?  Also, that's an entirely
+separate set of concerns.
+
+#### Example 2:  I have an operator that needs at least 3 of something to work.
+* With 2/3 replicas, warning alert after 60M.
+* With 1/3 replicas, warning alert after 10M.
+* With 0/3 replicas, warning alert after 5M.
+
+Q: How long should we wait until the above conditions rises to a 'critical'
+alert?
+
+A: It should never rise to the level of a critical alert, it's not critical.  If
+it was, this section would not apply.
+
+#### Example 3:  I have an operator that only needs 1 replica to function
+
+Well, if the cluster can upgrade with only 1 replica, and the the service is
+not unavailable despite other replicas being unavailable, this can probably
+be just an info-level alert.  We should have a "minor" alert, TBD.  But for now,
+"info" is the closest thing.
+
+#### Q: What if a particular condition will block upgrades?
+
+A: It's a warning level alert.
+
+
+### Alert Ownership
+
+Previously, the bulk of our alerting was handled directly by the monitoring
+team.  Initially, this gave use some indication into platform status without
+much effort by each component team.  As we mature our alerting system, we should
+ensure all teams take ownership of alerts for their individual components.
+
+Going forward, teams will be expected to own alerting rules within their own
+repositories.  This will reduce the burden on the monitoring team and better
+enable component owners to control their alerts.
+
+
+### Documentation Required
+
+Every alert needs to have a URL to relevant documentation to resolve the issue.
+
+If there's not documentation, there shouldn't be an alert.
+
+Sometimes an alert's impact is not obvious.  We should state the impact of the
+alert.  Don't expect users to understand the context, we have too many moving
+pieces.
+
+
+### Open Questions [optional]
+
+Frankly, having just "Critical" and "Warning" is not helpful.
+
+We really need "Critical", "Major", "Minor" and "Info."  Can/Should we do this?
+
+### Test Plan
+
+## Implementation History
+
+## Drawbacks
+
+People might have rules around the existing broken alerts.  They will have to
+change some of these rules.
+
+## Alternatives
+
+Document policies, but not make any backwards-incompatible changes to the
+existing alerts and only apply the policies to new alerts.
```

Comment:
Realizing you're listing this because it's in the template, but I feel compelled to say it is not an acceptable alternative from SRE standpoint.  We have to do two things:  1) improve existing noise 2) put in place a process (this doc) to ensure we don't regress.